app/src/main/java/org/p2p/wallet/history/ui/history/HistoryPresenter.kt

```diff
@@ -81,15 +86,19 @@ class HistoryPresenter(
 
     private suspend fun fetchHistory(isRefresh: Boolean = false) {
         try {
+            val moonpayItems = sellInteractor.loadUserSellTransactionsDetails()
+                .map { HistoryItem.MoonpayTransactionItem(it) }
+            moonpayTransactions.clear()
+            moonpayTransactions.addAll(moonpayItems)
             val fetchedItems = historyInteractor.loadTransactions(isRefresh)
             transactions.addAll(fetchedItems)
-            view?.showHistory(transactions)
+            view?.showHistory(transactions, moonpayItems)
```

Comment:
I would split the requests into separate suspend functions and call them here. 
This function is getting complex

Reply:
i'll do it in a separate task, history fragment is a mess that i will work on later